lib/rigtorp/SPSCQueue/include/rigtorp/SPSCQueue.h

```diff
@@ -151,7 +151,7 @@ template <typename T, typename Allocator = std::allocator<T>> class SPSCQueue {
     return try_emplace(std::forward<P>(v));
   }
 
-  T *front() noexcept {
+  T *front() const noexcept {
```

Comment:
Please strictly avoid to silently patch external code!! Instead file a PR for the external project if desired or if the signature is wrong.

Note: I didn't check if this change is reasonable or not.

Reply:
I think it's reasonable since it just returns the next element in the queue without popping it.

@xerus2000 Can you file an upstream PR?

Reply:
You are right, I also looked upstream and submitted a PR: https://github.com/rigtorp/SPSCQueue/pull/20
I've reverted the change for now, update coming soon.

Reply:
And I have been enlightened why it isn't const: https://github.com/rigtorp/SPSCQueue/issues/19